rust/tvm-rt/src/ndarray.rs

```diff
@@ -101,6 +101,21 @@ impl NDArrayContainer {
                 .cast::<NDArrayContainer>()
         }
     }
+
+    pub fn as_mut_ptr<'a>(object_ptr: &ObjectPtr<NDArrayContainer>) -> *mut NDArrayContainer
```

Comment:
This seems legit though some of the other functions on NDArrayContainer seem concerning from a rust memory model perspective.

`pub fn leak<'a>(object_ptr: ObjectPtr<NDArrayContainer>) -> &'a mut NDArrayContainer` seems like it can be used to hand out multiple &mut NDArrayContainer and should be considered unsafe.

Reply:
@robo-corg I think this is modeled after the other `leak` family of functions in Rust which take ownership https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak.

I admit this code is very tricky and scary it might be worth opening up follow up issue.